src/main/java/seedu/staffsnap/model/applicant/Applicant.java

```diff
@@ -179,12 +179,13 @@ public void addInterview(Interview interviewToAdd) {
     }
 
     /**
-     * Deletes an interview to an Applicant.
+     * Deletes an interview of an Applicant.
      *
-     * @param interviewToDelete the interview to delete from the Applicant
+     * @param interviewToDelete the interview to delete to the Applicant
      */
     public void deleteInterview(Interview interviewToDelete) {
         interviews.remove(interviewToDelete);
+        Collections.sort(interviews);
```

Comment:
Change interviews to priority queue to speed up.

Reply:
Great suggestion! I will update the Interview `List` into a `PriorityQueue` when the the sorting creates a bottleneck on performance.

I ran a benchmark testing for the sorting of a length 100 interview list and, as it now stands, it takes a negligible 25000 nanoseconds or 0.000025 seconds to execute. 

Since it is highly unlikely that any applicant will have more than 100 interviews, I see no immediate practical benefit in updating the underlying data structure. 

This is also in line with the suggestion in CS2103T's course website to _Avoid Premature Optimizations_.
Link: https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-premature-optimizations

What do you think?

Reply:
If you use a pq, then you won't need to remember to manually sort it when adding extra methods, reducing cleanup required and potential bugs, making your code more extensible :) Abstraction is an important principle of OOP

Reply:
> Change interviews to priority queue to speed up.

My initial comment was in response to your concern about _speed_. But since it appears you are now describing to a separate unrelated perspective, allow me to address those concerns as well.

You are bringing up excellent points 😄 Abstraction is indeed the cornerstone of OOP. In fact, that is the very reason why it might not be wise to concretize our interview structure this early on in the development phase. By preserving the `List` interface, it provides us with more abstraction, making the code more flexible and resilient to changes.

Suppose we wish to implement a function that sorts the interview list down the road. A `List` would allow us to sort by various comparators, whereas a `PriorityQueue` would have it's comparator concretized, restricting extensibility.